entrypoint/help_utils.go

```diff
@@ -0,0 +1,155 @@
+package entrypoint
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"regexp"
+	"strings"
+	"text/tabwriter"
+	"text/template"
+	"unicode"
+)
+
+const HELP_TEXT_LINE_WIDTH = 80
+
+const CLI_APP_HELP_TEMPLATE = `Usage: {{.HelpName}} {{range $index, $option := .VisibleFlags}}[{{$option.GetName | PrefixedFirstFlagName}}] {{end}}{{if .Commands}}command [options]{{end}} {{if .ArgsUsage}}{{.ArgsUsage}}{{else}}[args]{{end}}{{if .Description}}
+
+{{.Description}}{{end}}{{if .Commands}}
+
+Commands:
+
+{{range .Commands}}{{if not .HideHelp}}   {{join .Names ", "}}{{ "\t"}}{{.Usage}}{{ "\n" }}{{end}}{{end}}{{end}}{{if .VisibleFlags}}{{end}}
+`
+
+const CLI_COMMAND_HELP_TEMPLATE = `Usage: {{if .UsageText}}{{.UsageText}}{{else}}{{.HelpName}}{{if .VisibleFlags}} [options]{{end}} {{if .ArgsUsage}}{{.ArgsUsage}}{{else}}[args]{{end}}{{end}}{{if .Description}}
+
+{{.Description}}{{end}}{{if .VisibleFlags}}
+
+Options:
+
+   {{range .VisibleFlags}}{{.}}
+   {{end}}{{end}}
+`
+
+// HelpPrinter that will wrap the text at
+// HELP_TEXT_LINE_WIDTH characters, while preserving
+// indentation and table tabs. Currently only works with
+// tables delimited by `\t` and with only 2 columns.
+func WrappedHelpPrinter(out io.Writer, templateString string, data interface{}) {
+	funcMap := template.FuncMap{
+		"join":                  strings.Join,
+		"PrefixedFirstFlagName": PrefixedFirstFlagName,
+	}
+
+	templ := template.Must(template.New("help").Funcs(funcMap).Parse(templateString))
+	rendered := bytes.NewBufferString("")
+	err := templ.Execute(rendered, data)
+	if err != nil {
+		return
+	}
+
+	writer := tabwriter.NewWriter(out, 0, 8, 2, ' ', 0)
+	for _, line := range strings.Split(rendered.String(), "\n") {
+		tab := HelpTableAwareDetermineIndent(line, "\t+")
+		wrappedLine := TabAwareWrapText(line, HELP_TEXT_LINE_WIDTH, tab)
+		fmt.Fprintln(writer, wrappedLine)
+	}
+	writer.Flush()
+}
+
+// Similar functionality to regexp.Split, but returns the delimited strings
+// with the trailing delimiter appended to it.
+// Example:
+//    re := regexp.MustCompile("\\s+")
+//    text := "one two three"
+//    out := SplitKeepDelimiter(re, text)
+//    out == ["one ", "two ", "three"]
+func SplitKeepDelimiter(re *regexp.Regexp, str string) (out []string) {
```

Comment:
> Why declare out in the return value?

No strong reason, just that I noticed a few places in sample go code preferring to declare in return value if you are just going to `var out`. Is it generally preferable to declare return vars in the function?

> Also, does SplitAfterN not do the same thing?

`strings.SplitAfter` does do the same thing, but it requires a specific substring. Here, we need to split on regex because the separator is volatile (at least one whitespace char) and so there is no exact match string that can be used.

Reply:
> Is it generally preferable to declare return vars in the function?

I don't know what "idiomatic" Go code does (TBH, I'm not sure it's mature enough of a language for that to be well defined), but I find it easier to read and reason about the code if all variable declarations are in the body of the method, rather than some there and some in the return value. 

> strings.SplitAfter does do the same thing, but it requires a specific substring. Here, we need to split on regex because the separator is volatile (at least one whitespace char) and so there is no exact match string that can be used.

Ah, gotcha.

Reply:
Renamed this func to `RegexpSplitAfter` to make it clearer that this is the regex version of `strings.SplitAfter`.